energysmartbridge/src/Modules/MQTTModule.cs

```diff
@@ -246,53 +230,53 @@ private void PublishWaterHeater(WaterHeaterInput waterHeater)
+            PublishAsync($"{Global.mqtt_discovery_prefix}/binary_sensor/{waterHeater.DeviceText}/leak/config",
                 JsonConvert.SerializeObject(waterHeater.ToLeakConfig()));
 
-            PublishAsync($"{Global.mqtt_discovery_prefix}/sensor/{waterHeater.DeviceText}/masterdispfail/config",
+            PublishAsync($"{Global.mqtt_discovery_prefix}/binary_sensor/{waterHeater.DeviceText}/masterdispfail/config",
                 JsonConvert.SerializeObject(waterHeater.ToMasterDispFailConfig()));
 
-            PublishAsync($"{Global.mqtt_discovery_prefix}/sensor/{waterHeater.DeviceText}/compsensorfail/config",
+            PublishAsync($"{Global.mqtt_discovery_prefix}/binary_sensor/{waterHeater.DeviceText}/compsensorfail/config",
                 JsonConvert.SerializeObject(waterHeater.ToCompSensorFailConfig()));
 
-            PublishAsync($"{Global.mqtt_discovery_prefix}/sensor/{waterHeater.DeviceText}/syssensorfail/config",
+            PublishAsync($"{Global.mqtt_discovery_prefix}/binary_sensor/{waterHeater.DeviceText}/syssensorfail/config",
                 JsonConvert.SerializeObject(waterHeater.ToSysSensorFailConfig()));
 
-            PublishAsync($"{Global.mqtt_discovery_prefix}/sensor/{waterHeater.DeviceText}/systemfail/config",
+            PublishAsync($"{Global.mqtt_discovery_prefix}/binary_sensor/{waterHeater.DeviceText}/systemfail/config",
                 JsonConvert.SerializeObject(waterHeater.ToSystemFailConfig()));
 
             PublishAsync($"{Global.mqtt_discovery_prefix}/sensor/{waterHeater.DeviceText}/faultcodes/config",
                 JsonConvert.SerializeObject(waterHeater.ToFaultCodesConfig()));
+
+            PublishAsync($"{Global.mqtt_discovery_prefix}/sensor/{waterHeater.DeviceText}/signalstrength/config",
+                JsonConvert.SerializeObject(waterHeater.ToSignalStrengthConfig()));
         }
 
         private void PublishWaterHeaterState(WaterHeaterInput waterHeater)
         {
             PublishAsync(waterHeater.ToTopic(Topic.maxsetpoint_state), waterHeater.MaxSetPoint.ToString());
             PublishAsync(waterHeater.ToTopic(Topic.setpoint_state), waterHeater.SetPoint.ToString());
 
-            string ha_mode = "off";
-            switch(waterHeater.Mode) {
-                case "Efficiency":
-                    ha_mode = "heat_pump";
-                    break;
-                case "Hybrid":
-                    ha_mode = "eco";
-                    break;
-                case "Electric":
-                    ha_mode = "electric";
-                    break;
-                default:
-                    break;
-            }
+            string ha_mode = waterHeater.Mode switch
+            {
+                "Efficiency" => "heat_pump",
+                "Hybrid" => "eco",
+                "Electric" => "electric",
+                "Vacation" => "off",
+                _ => "unknown",
```

Comment:
I dont think unknown is a supported state by HA. Ive not played around with HA much to know what will happen if we send an unsupported state which is why my prior code just fellback to off rather then something else. But open to change this if its safe to do so